config/config_mgmt.py

```diff
@@ -0,0 +1,841 @@
+
+        Returns:
+            (bool): True, if given key is present.
+        '''
+        self.sysLog(msg='Check Key in Asic DB: {}'.format(key))
+        try:
+            # chk key in ASIC DB
+            if db.exists('ASIC_DB', key):
+                return True
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            raise(e)
+
+        return False
+
+    def _checkNoPortsInAsicDb(self, db, ports, portMap):
+        '''
+        Check ASIC DB for PORTs in port List
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): List of ports
+            portMap (dict): port to OID map.
+
+        Returns:
+            (bool): True, if all ports are not present.
+        '''
+        try:
+            # connect to ASIC DB,
+            db.connect(db.ASIC_DB)
+            for port in ports:
+                key = self.oidKey + portMap[port]
+                if self._checkKeyinAsicDB(key, db) == True:
+                    return False
+
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            return False
+
+        return True
+
+    def _verifyAsicDB(self, db, ports, portMap, timeout):
+        '''
+        Verify in the Asic DB that port are deleted, Keep on trying till timeout
+        period.
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): port list to check in ASIC DB.
+            portMap (dict): oid<->port map.
+            timeout (int): timeout period
+
+        Returns:
+            (bool)
+        '''
+        self.sysLog(doPrint=True, msg="Verify Port Deletion from Asic DB, Wait...")
+        try:
+            for waitTime in range(timeout):
+                self.sysLog(logLevel=syslog.LOG_DEBUG, msg='Check Asic DB: {} \
+                    try'.format(waitTime+1))
+                # checkNoPortsInAsicDb will return True if all ports are not
+                # present in ASIC DB
+                if self._checkNoPortsInAsicDb(db, ports, portMap):
+                    break
+                tsleep(1)
+
+            # raise if timer expired
+            if waitTime + 1 == timeout:
+                self.sysLog(syslog.LOG_CRIT, "!!!  Critical Failure, Ports \
+                    are not Deleted from ASIC DB, Bail Out  !!!", doPrint=True)
+                raise(Exception("Ports are present in ASIC DB after {} secs".\
+                    format(timeout)))
+
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            raise e
+
+        return True
+
+    def breakOutPort(self, delPorts=list(), portJson=dict(), force=False, \
+            loadDefConfig=True):
+        '''
+        This is the main function for port breakout. Exposed to caller.
+
+        Parameters:
+            delPorts (list): ports to be deleted.
+            addPorts (list): ports to be added.
```

Comment:
Agree with Renuka. Please remove this line, as it is no longer a parameter tot he function.

Reply:
fixed